lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -81,8 +81,50 @@
     <cmdline><%= @cmd_line %></cmdline>
 <%- if @dtb -%>
     <dtb><%= @dtb %></dtb>
+<% end -%>
+<%- unless @sysinfo.empty? -%>
+    <smbios mode='sysinfo'/>
 <% end -%>
   </os>
+<%- unless @sysinfo.empty? -%>
+  <sysinfo type='smbios'>
+  <%- if @sysinfo.has_key?('bios') -%>
+    <bios>
+    <%- @sysinfo[:bios].each do |key, value| -%>
+      <entry name='<%= key %>'><%= value %></entry>
+    <% end -%>
+    </bios>
+  <% end -%>
+  <%- if @sysinfo.has_key?('system') -%>
+    <system>
+    <%- @sysinfo[:system].each do |key, value| -%>
+      <entry name='<%= key %>'><%= value %></entry>
+    <% end -%>
+    </system>
+  <% end -%>
+  <%- if @sysinfo.has_key?('base_board') -%>
+    <baseBoard>
+    <%- @sysinfo[:base_board].each do |key, value| -%>
+      <entry name='<%= key %>'><%= value %></entry>
+    <% end -%>
+    </baseBoard>
+  <% end -%>
+  <%- if @sysinfo.has_key?('chassis') -%>
+    <chassis>
+    <%- @sysinfo[:chassis].each do |key, value| -%>
+      <entry name='<%= key %>'><%= value %></entry>
+    <% end -%>
+    </chassis>
+  <% end -%>
+  <%- if @sysinfo.has_key?('oem_strings') -%>
+    <oemStrings>
+    <%- @sysinfo[:oem_strings].each do |value| -%>
+      <entry><%= value %></entry>
+    <% end -%>
+    </oemStrings>
+  <% end -%>
+  </sysinfo>
+<% end -%>
```

Comment:
I'm wondering if there is some restriction with erb and string vs symbols conversion, I'd suggest using the same rather than switching between when you check the key and then attempt to access it.

Reply:
I assumed they are the same and just found out they are not, especially when used as key in a hash. Now I do understand your comment. :)